chromadb/telemetry/product/events.py

```diff
@@ -80,17 +82,36 @@ def __init__(
         with_embeddings: int,
         with_metadata: int,
         with_documents: int,
+        batch_size: int = 1,
     ):
         super().__init__()
         self.collection_uuid = collection_uuid
         self.update_amount = update_amount
         self.with_embeddings = with_embeddings
         self.with_metadata = with_metadata
         self.with_documents = with_documents
+        self.batch_size = batch_size
 
+    @property
+    def batch_key(self) -> str:
+        return self.collection_uuid + self.name
+
+    def batch(self, other: "ProductTelemetryEvent") -> "CollectionUpdateEvent":
+        if not self.batch_key == other.batch_key:
+            raise ValueError("Cannot batch events")
+        other = cast(CollectionUpdateEvent, other)
+        total_amount = self.update_amount + other.update_amount
+        return CollectionUpdateEvent(
+            collection_uuid=self.collection_uuid,
+            update_amount=total_amount,
+            with_documents=self.with_documents + other.with_documents,
+            with_metadata=self.with_metadata + other.with_metadata,
+            with_embeddings=self.with_embeddings + other.with_embeddings,
+            batch_size=self.batch_size + other.batch_size,
+        )
 
 class CollectionQueryEvent(ProductTelemetryEvent):
-    max_batch_size: ClassVar[int] = 20
+    max_batch_size: ClassVar[int] = 1000
```

Comment:
are we going to lose a ton of data on top of funnel users just prototyping ?

Reply:
The batching system is designed to always send the first event of each batch_key, only batching if an event has been seen before.

Sometime in the next couple weeks I'll land a better fix where batch_size increases as the number of events goes up but this should do for now I think.